src/main/scala/mesosphere/marathon/core/readiness/ReadinessCheck.scala

```diff
@@ -35,6 +38,14 @@ object ReadinessCheck {
     case object HTTPS extends Protocol
   }
 
-  implicit val readinessCheckValidator = validator[ReadinessCheck] { rc =>
-  }
+  def readinessCheckValidator(portDefinitions: Seq[PortDefinition]): Validator[ReadinessCheck] =
+    validator[ReadinessCheck] { rc =>
+      rc.name is notEmpty
+      rc.path is notEmpty
+      rc.portName is notEmpty
+      rc.portName is oneOf(portDefinitions.flatMap(_.name): _*)
+      rc.interval.toSeconds should be > 0L
+      rc.timeout.toSeconds should be > 0L
```

Comment:
One additional thing that would make sense to check: timeout < interval
(so we can make sure, there is only one check per task at a time.